viewing_party/party.py

```diff
@@ -1,23 +1,183 @@
 # ------------- WAVE 1 --------------------
 
-def create_movie(title, genre, rating):
-    pass
+def create_movie(movie_title, genre, rating):
+    if movie_title == None or genre == None or rating == None:
+        return None
+    movie = {"title":movie_title, "genre":genre, "rating":rating}
+    return movie
 
+def add_to_watched(user_data, movie):
+    if movie["title"] not in user_data["watched"]:
+        user_data["watched"].append(movie)
+    return user_data
+    
+def add_to_watchlist(user_data, movie):
+    if movie["title"] not in user_data["watchlist"]:
+        user_data["watchlist"].append(movie)
+    return user_data
+    
+def watch_movie(user_data, movie_title):
+    for movie in user_data["watchlist"]:
+        if movie_title == movie["title"] and movie_title not in user_data["watched"]:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+    return user_data
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    rating = 0.0
+    count = 0
+
+    if not user_data:
+        return None
+
+    for movies in user_data.values():
+        if not movies:
+            average_rating = 0.0
+            return average_rating
+        
+        for movie in movies:
+            count += 1
+            rating += movie["rating"]
+    
+    rating = rating/count

```

Comment:
Nitpick: Incorrect white spacing around operator

Consider using the length of watched and remove `count` variable.
```suggestion
    rating = rating / len(user_data["watched"])
```